packages/server/src/question/question.service.ts

```diff
@@ -137,6 +141,35 @@ export class QuestionService {
       if (question.group) question.group = null;
       else question.groupId = null;
     }
+
+    // For Queue Chats
+    try {
+      if (isResolving) {
+        // Save chat metadata in database (if messages were exchanged)
+        await this.queueChatService.endChat(question.queueId, question.id);
+      } else if (isBecomingClosedFromWaiting) {
+        // Don't save chat metadata in database
+        await this.queueChatService.clearChat(question.queueId, question.id);
+      } else if (isFirstHelped) {
+        // Create chat metadata in Redis
+        await this.queueChatService.createChat(
+          question.queueId,
+          question.taHelped,
+          question,
+        );
+      }
```

Comment:
```suggestion
      if (isResolving) {
        // Save chat metadata in database (if messages were exchanged)
        await this.queueChatService.endChat(question.queueId, question.id);
      } else if (newStatus in ClosedQuestionStatus) {
        // Don't save chat metadata in database
        await this.queueChatService.clearChat(question.queueId, question.id);
      } else if (isFirstHelped) {
        // Create chat metadata in Redis
        await this.queueChatService.createChat(
          question.queueId,
          question.taHelped,
          question,
        );
      }
```

Hmmm, I realise that there might be a case where a question gets deleted from limbo status or something and it ends up maybe not clearing the chat. So I'm thinking it might be better to just check if newStatus is in ClosedQuestionStatus

Reply:
i am going to implement these changes real quick